src/AddressRegistry.Api.Legacy/AddressMatch/Matching/Sanitizer.cs

```diff
@@ -5,376 +5,599 @@ namespace AddressRegistry.Api.Legacy.AddressMatch.Matching
+                    formattedParts.HouseNumber = possibleNumberInStreet;
                 }
-                //huisnummer met niet-numeriek bisnummer als suffix in straatnaam
-                if (new Regex(" [0-9]+[a-zA-Z]$").IsMatch(streetName))
+            }
+
+            // huisnummer met niet-numeriek bisnummer als prefix in straatnaam
+            // TODO: [0-9]+ ?
+            if (new Regex("[0-9][a-zA-Z] ").IsMatch(streetName) && !new Regex("[0-9]e ").IsMatch(streetName))
+            {
+                if (possibleNumberInStreet != null
+                    && int.TryParse(possibleNumberInStreet, out hnr)
+                    && hnr != 0
+                    // TODO: [0-9]+ ?
+                    && new Regex("[0-9][a-zA-Z]$").IsMatch(requestStraatnaamWithoutStraatnaam))
                 {
-                    if (FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)) != null
-                        && Int32.TryParse(FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)), out hnr)
-                        && hnr != 0
-                        && new Regex("[0-9][a-zA-Z]$").IsMatch(requestStraatnaamWithoutStraatnaam))
-                    {
-                        houseNumber0 = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
-                        rrindex0 = FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam)).ToUpper();
-                        //TODO Stif: ask business, why only here warning?
-                        Warnings.AddWarning("3", "Niet-numeriek 'Huisnummer' in 'Straatnaam' gevonden.");
-                    }
+                    formattedParts.HouseNumber = possibleNumberInStreet;
+                    formattedParts.RrIndex = possibleIndexInStreet.ToUpper();
                 }
             }
 
-            /* huisnummer zonder RR-index */
-            /* Het huisnummer wordt een CRAB-huisnummer. Er is geen CRAB-subadres. */
-            /********************************************************/
-            //volledig numerisch
-            if (new Regex("^[0-9]+$").IsMatch(houseNumber0) && rrindex0 == "0000")
+            // huisnummer als suffix in straatnaam
+            if (new Regex(" [0-9]+$").IsMatch(streetName))
             {
-                return new List<HouseNumberWithSubaddress> { new HouseNumberWithSubaddress(RemovePrecedingZeros(houseNumber0), null, null) };
+                formattedParts.HouseNumber = FormatRrHouseNumber(new Regex(" [0-9]+$").Match(streetName).Value);
+
+                Warnings.AddWarning("2", "'Huisnummer' in 'Straatnaam' gevonden.");
             }
-            //niet-numeriek bisnummer: enkele letter
-            if (rrindex0 == "0000" && new Regex(@"^[0-9]+\s*[a-zA-Z]$").IsMatch(houseNumber0))
+
+            // huisnummer met niet-numeriek bisnummer als suffix in straatnaam
+            // TODO: We dont care about excluding [0-9]e anymore?
+            if (new Regex(" [0-9]+[a-zA-Z]$").IsMatch(streetName))
```

Comment:
We dont care about excluding `[0-9]e` anymore? We did before:

`if (new Regex("[0-9][a-zA-Z] ").IsMatch(streetName) && !new Regex("[0-9]e ").IsMatch(streetName))`

Reply:
Checking with another developer